File: php/p005.php
function gcd($a, $b) {
	while ($b != 0) {
		$c = $a % $b;
		$a = $b;
		$b = $c;
	}
	return $a;
}

function main() {
	$count = 1;
	for ($i = 1; $i <= 20; $i++) {
		$count *= floor($i / gcd($i, $count));
	}
	return $count;
}
echo main();

File: javasec/p005.java
package javasec;

public class p005 {
	public int gcd(int a, int b) {
		while (b != 0) {
			int c = a % b;
			a = b;
			b = c;
		}
		return a;
	}
	public int lcm(int a, int b) {
		return a / gcd(a, b) * b;
	}

	public int run() {
		int count = 1;
		for (int i = 1; i < 21; i++) {
			count *= (int)Math.floor(i / gcd(i, count));
		}
		return count;
	}
	public static void main(String[] args) {
		System.out.println(new p005().run());
	}
}
